galaxy_importer/collection.py

```diff
@@ -173,12 +232,105 @@ def _load_collection_manifest(self):
+            return data
+
+    def _load_collection_file_manifest(self, path_prefix, file_manifest_file):
+        """Load CollectionArtifactFileManifest data from file_manifest_file
+
+        Args:
+            path_prefix (str): Any file path prefix we need to add to file paths in the
+                CollectionArtifactFile artifact_file
+            file_manifest_file (CollectionArtifactFile): object with info (name, path) of the files
+                in the artifact. ie, "FILES.json"
+
+        Raises:
+            ManifestValidationError: If there are any errors loading, parsing,
+                deserializing, or validating the data in file_manifest_file
+
+        Returns:
+            CollectionArtifactFileManifest: The data from file_manifest_file
+
+        """
+        default_logger.debug('file_manifest_file: %s', file_manifest_file)
+
+        check_artifact_file(path_prefix=path_prefix, artifact_file=file_manifest_file)
+
+        files_manifest_file = os.path.join(path_prefix, file_manifest_file.name)
+        default_logger.debug('files_manifest_file: %s', files_manifest_file)
+
+        with open(files_manifest_file, 'r') as f:
+            try:
+                file_manifest = schema.CollectionArtifactFileManifest.parse(f.read())
+            except ValueError as e:
+                # TODO: Should this be a subclass of ValueError?
+                raise exc.ManifestValidationError(str(e))
+
+        return file_manifest
+
+    def _check_file_manifest(self, path_prefix, file_manifest, file_manifest_name):
+        """Check the file content described in file_manifest
+
+        Check the chksums for files.
+        Check for any missing files.
+        Check for any missing dirs.
+
+        Args:
+            path_prefix (str): Any file path prefix we need to add to file paths in the
+                CollectionArtifactFile artifact_file
+            file_manifest (CollectionArtifactFileManifest): Object with list of
+                CollectionArtifactFile items. The CollectionArtifactFile data
+                is used to check and validate each file.
+            file_manifest_name (str): The name of the file manifest (ie, "FILES.json")
+
+        Raises:
+            CollectionArtifactFileNotFound: If artifact_file is not found on the file system.
+            CollectionArtifactFileChecksumError: If the sha256sum of the on disk
+                artifact_file contents does not match artifact_file.chksum_sha256.
+
+        Returns:
+            bool: All the items in file_manifest were found and valid
+        """
+
+        for artifact_file in file_manifest.files:
+            if artifact_file.ftype != 'file':
+                continue
+
+            check_artifact_file(path_prefix=path_prefix,
+                                artifact_file=artifact_file)
+
+        # check the extract archive for any extra files.
+        ftw = FileWalker(collection_path=path_prefix)
```

Comment:
not sure if ftw is a common acronym, but `filewalker` would be a standard name as an instance of the class

Reply:
Agreed, lowercase 'filewalker' is better. Will update.

ftw is semi-common (there is a ftw() in std libc).

(I think I was thinking it would collide with `filewalker` module import, but never added a 'filewalker' module and didn't update the name)

Reply:
updated